rustls/src/verify.rs

```diff
@@ -539,24 +540,28 @@ fn prepare<'a, 'b>(
 }
 
 /// A `ClientCertVerifier` that will ensure that every client provides a trusted
-/// certificate, without any name checking.
+/// certificate, without any name checking. Optionally, client certificates will
+/// have their revocation status checked using the DER encoded CRLs provided.
 pub struct AllowAnyAuthenticatedClient {
     roots: RootCertStore,
     subjects: Vec<DistinguishedName>,
+    crls: Vec<Vec<u8>>,
```

Comment:
I'll probably put up a branch extending https://github.com/rustls/webpki/pull/92 with an owned representation for comparison purposes anyway. I think it will be required one way or the other for https://github.com/rustls/webpki/issues/80 and https://github.com/rustls/webpki/issues/81

Reply:
I think even if we're going to keep the CRL data around as a `Vec<u8>` we should wrap that in a newtype wrapper (like `PrivateKey` and `Certificate`) to make usage somewhat more intentional.

Reply:
Good idea.

Reply:
Added the newtype wrapper in my recent push.

Reply:
> The intuition here is that going from DER to the CRL view over the DER is cheap (especially since signature validation isn't done until a cert with an issuer matching the CRL is consulted for a serial), but that should be verified with benchmarking (I will pursue that separately).

Started on benchmarking: https://github.com/rustls/webpki/pull/94

Initial results show parsing the CRL and finding serials in the CRL are roughly equivalent in runtime, with both seeming fairly quick from my perspective.

Reply:
I guess the next question is, how does this stuff compare in performance to the other things a webpki client certificate verification has to run through? That is, what is the impact on this relative to the whole operation?

Reply:
> I guess the next question is, how does this stuff compare in performance to the other things a webpki client certificate verification has to run through?

Probably the most substantial comparison point would be to path building in general, though my expectation there is that signature validation will be the dominant factor. I'll give this some thought.

> Initial results show parsing the CRL and finding serials in the CRL are roughly equivalent in runtime, with both seeming fairly quick from my perspective.

Just noting that this has an obvious explanation I overlooked :facepalm: - When we parse CRLs we're also walking the entire set of revoked certificates to detect errors early.

Before I start on a more time intensive side quest to benchmark path building I'm going to rework the code under benchmark to avoid doing that up-front check so we can have a better sense of the speed of just parsing the base CRL vs parsing/searching the revoked certificates.

Reply:
>  I'm going to rework the code under benchmark to avoid doing that up-front check so we can have a better sense of the speed of just parsing the base CRL vs parsing/searching the revoked certificates.

Pushed that change to https://github.com/rustls/webpki/pull/94 and the results now clearly show that parsing the base CRL is very fast relative to parsing and searching its revoked cert contents.